src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -982,105 +1139,126 @@ namespace EnergyPlus {
+                                                                                           thisPLHP.name,
                                                                                            DataLoopNode::NodeType_Water,
                                                                                            DataLoopNode::NodeConnectionType_Inlet,
                                                                                            flowPath1,
                                                                                            DataLoopNode::ObjectIsNotParent);
-                        thisWWHP.loadSideNodes.outlet = NodeInputManager::GetOnlySingleNode(loadSideOutletNodeName,
+                        thisPLHP.loadSideNodes.outlet = NodeInputManager::GetOnlySingleNode(loadSideOutletNodeName,
                                                                                             nodeErrorsFound,
                                                                                             cCurrentModuleObject,
-                                                                                            thisWWHP.name,
+                                                                                            thisPLHP.name,
                                                                                             DataLoopNode::NodeType_Water,
                                                                                             DataLoopNode::NodeConnectionType_Outlet,
                                                                                             flowPath1,
                                                                                             DataLoopNode::ObjectIsNotParent);
-                        thisWWHP.sourceSideNodes.inlet = NodeInputManager::GetOnlySingleNode(sourceSideInletNodeName,
+                        int condenserNodeType = 0;
+                        int condenserNodeConnectionType_Inlet = 0;
+                        int condenserNodeConnectionType_Outlet = 0;
+                        if (condenserType == "WATERSOURCE") {
+                            thisPLHP.waterSource = true;
+                            condenserNodeType = DataLoopNode::NodeType_Water;
+                            condenserNodeConnectionType_Inlet = DataLoopNode::NodeConnectionType_Inlet;
+                            condenserNodeConnectionType_Outlet = DataLoopNode::NodeConnectionType_Outlet;
+                        } else if (condenserType == "AIRSOURCE") {
+                            thisPLHP.airSource = true;
+                            condenserNodeType = DataLoopNode::NodeType_Air;
+                            condenserNodeConnectionType_Inlet = DataLoopNode::NodeConnectionType_OutsideAir;
+                            condenserNodeConnectionType_Outlet = DataLoopNode::NodeConnectionType_OutsideAir;
```

Comment:
YES this way you can still have single calls to GetOnlySingleNode.  This is the stuff.